### PR TITLE
Adds Drawer to NavDrawer

### PR DESCRIPTION
**Pull request by mltejera**
This is the first of a few major changes to NavDrawer, which is adding a Nav provider to the drawer.

The idea is consumers can skip a layer of complexity and interact with NavDrawer the way they do with both Nav and Drawer components.

Still to add in future PRs:
- General styling
- Tabster functionality to make keyboarding arrow behavior
- Custom Style Hooks so we can change the styles of components within the nav

PFA: Visuals don't change except to verify props for drawer pass through.

image

Chips away at #26649

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/56422e2212b44e023d187fc924cd14aa68ecda8e'>56422e2212b44e023d187fc924cd14aa68ecda8e</a></sub>
<div id="bundleSizeComment9423"/>

File: packages/react-components/react-nav-preview/src/components/NavDrawer/NavDrawer.tsx [hunk 1]
```diff
@@ -5,18 +5,21 @@ import { useNavDrawer_unstable } from './useNavDrawer';
 import { renderNavDrawer_unstable } from './renderNavDrawer';
 import { useNavDrawerStyles_unstable } from './useNavDrawerStyles.styles';
 import type { NavDrawerProps } from './NavDrawer.types';
+import { useNavContextValues_unstable } from '../useNavContextValues';
+import { NavState } from '../Nav/Nav.types';
```
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/NavDrawer.tsx:9]:
```suggestion
import { useNavContextValues_unstable } from '../useNavContextValues';
import { NavState } from '../Nav/Nav.types';
import type { NavDrawerProps } from './NavDrawer.types';
```

File: packages/react-components/react-nav-preview/src/components/NavDrawer/useNavDrawer.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import * as React from 'react';
-import { getIntrinsicElementProps, slot } from '@fluentui/react-utilities';
 import type { NavDrawerProps, NavDrawerState } from './NavDrawer.types';
+import { useInlineDrawer_unstable } from '@fluentui/react-drawer';
+import { useNav_unstable } from '../Nav/useNav';
```
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/useNavDrawer.ts:4]:
```suggestion
import { useInlineDrawer_unstable } from '@fluentui/react-drawer';
import { useNav_unstable } from '../Nav/useNav';
import type { NavDrawerProps, NavDrawerState } from './NavDrawer.types';
```

File: packages/react-components/react-nav-preview/src/components/NavDrawer/useNavDrawerStyles.styles.ts [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import { makeStyles, mergeClasses } from '@griffel/react';
 import type { SlotClassNames } from '@fluentui/react-utilities';
-import type { NavDrawerSlots, NavDrawerState } from './NavDrawer.types';
+import type { NavDrawerState } from './NavDrawer.types';
+import { InlineDrawerSlots, useInlineDrawerStyles_unstable } from '@fluentui/react-drawer';
```
Comment (sopranopillow) [inline packages/react-components/react-nav-preview/src/components/NavDrawer/useNavDrawerStyles.styles.ts:4]:
```suggestion
import { InlineDrawerSlots, useInlineDrawerStyles_unstable } from '@fluentui/react-drawer';
import type { NavDrawerState } from './NavDrawer.types';
```

I'm also missing a type import in line 2 that should be at the bottom

**Pull request closed without merging**